SC2/GameCalcs.cpp

```diff
@@ -69,13 +70,14 @@ double CGameCalcs::CalculateGasIncomeRate2Near(size_t baseCount, size_t geyserBu
 	if(workerCount > 3*geyserBuildingCount)
 		workerCount = 3*geyserBuildingCount;
 
-	return (114/60.0) * (workerCount/3) + closeGeyserSumIncome[workerCount%3];
+	size_t incomeRateIndex = workerCount / geyserBuildingCount;
+	return geyserBuildingCount* closeGeyserSumIncome[incomeRateIndex] + (workerCount % geyserBuildingCount)*incomeRate[incomeRateIndex];
 }
 
 double CGameCalcs::CalculateGasIncomeRate1Near1Far(size_t baseCount, size_t geyserBuildingCount, size_t workerCount)
 {
-	static double closeGeyserSumIncome[] = {0.0, 38/60.0, 76/60.0, 114/60.0}; // Sum of income rates
-	static double farGeyserSumIncome[] = {0.0, 33/60.0, 67/60.0, 101/60.0, 114/60.0}; // Sum of income rates
+	static double closeGeyserSumIncome[] = {0.0, 56/60.0, 112/60.0, 158/60.0}; // Sum of income rates
+	static double farGeyserSumIncome[] = {0.0, 12/60.0, 23/60.0, 34/60.0, 45/60.0}; // Sum of income rates
```

Comment:
These numbers don't seem right. It suggests a geyser that's further from the base only mines at a quarter of the rate of a close geyser.

I think the farGeyserSumIncome should be something like:
{0.0, 48/60.0, 96/60.0, 144/60.0, 158/60.0};

Reply:
We'd also need to fix up the (114/60.0) in the return statement below.

Given these 1Near1Far & 2Far functions aren't used, I'd be inclined just to remove them rather than try to maintain unused calculations.

Reply:
Sure. Give me a couple of days to check the numbers again. I think I misunderstood "far" here.